x/evm/keeper/fees_test.go

```diff
@@ -270,22 +272,30 @@ func (suite *KeeperTestSuite) TestCheckSenderBalance() {
 // in one function and share a lot of the same setup.
 // In practice, the two tested functions will also be sequentially executed.
 func (suite *KeeperTestSuite) TestVerifyFeeAndDeductTxCostsFromUserBalance() {
-	hundredInt := sdkmath.NewIntFromBigInt(big.NewInt(5e18))
+	var initialBaseFee *big.Int
+	hundredInt := sdkmath.NewInt(100)
 	zeroInt := sdkmath.ZeroInt()
-	// oneIntOld := sdkmath.NewInt(1)
-	oneInt := sdkmath.NewInt(100000000000000)
-	fiveInt := sdkmath.NewInt(50000000000000)
-	fiftyInt := sdkmath.NewInt(50000000000000)
+	oneInt := sdkmath.OneInt()
+	fiveInt := sdkmath.NewInt(5)
+	fiftyInt := sdkmath.NewInt(50)
 
 	// should be enough to cover all test cases
-	initBalance := sdkmath.NewInt((ethparams.InitialBaseFee + 10) * 100000000)
+	initBalance := func() sdkmath.Int { return (sdkmath.NewIntFromBigInt(initialBaseFee).AddRaw(10)).MulRaw(105) }
 
+	testSetup := []struct {
+		decimals       uint32
+		initialBaseFee math.LegacyDec
+	}{
+		{evmtypes.Denom18Dec, sdkmath.LegacyNewDec(params.InitialBaseFee)},
+		{evmtypes.Denom6Dec, sdkmath.LegacyNewDec(1e13)},
```

Comment:
Since this code will be part of the OS probably it is better to have InitialBaseFee18 and InitialBaseFee6 in the params.

Reply:
this PR is specifically to run these tests with these baseFees that are equivalent in 6 and 18 dec. Not necessarily the default baseFee on both cases. But we can define these anyways in other PR